sympy/integrals/manualintegrate.py

```diff
@@ -137,6 +138,36 @@ def manual_diff(f, symbol):
                 return f.args[0] * manual_diff(f.args[1], symbol)
     return f.diff(symbol)
 
+def manual_subs(expr, *args):
+    """
+    A wrapper for `expr.subs(*args)` with additional logic for substitution
+    of invertible functions.
+    """
+    if len(args) == 1:
+        sequence = args[0]
+        if isinstance(sequence, (Dict, Mapping)):
+            sequence = sequence.items()
+        elif not iterable(sequence):
+            raise ValueError("Expected an iterable of (old, new) pairs")
+    elif len(args) == 2:
+        sequence = [args]
+    else:
+        raise ValueError("subs accepts either 1 or 2 arguments")
+
+    new_subs = []
+    for old, new in sequence:
+        if isinstance(old, sympy.log):
+            # If log(x) = y, then exp(a*log(x)) = exp(a*y)
+            # that is, x**a = exp(a*y). Replace nontrivial powers of x
+            # before subs turns them into `exp(y)**a`, but
+            # do not replace x itself yet, to avoid `log(exp(y))`.
+            a = sympy.Wild('a')
+            expr = expr.replace(old.args[0]**(1 + a),
```

Comment:
Why is this 1 + a?

Reply:
I want to match Pow(x, y) but not x itself (which would match with y=1). The parameter `exact=True`  restricts matches to nonzero values of Wild. So, `1+a` ensures the exponent is not `1`.

Reply:
This would probably work with `a` instead of  `1 + a` as `exp(log(x))` is automatically evaluated to `x`. On the other hand, it seems that `e.match(e**a)` would match any `e`. So `1 + a` with `exact=True` might save some time by skipping over non-Pow nodes.